src/pages/events/[eventId]/EventUnRegisterModal.jsx

```diff
@@ -0,0 +1,52 @@
+import { useState } from "react";
+import { Modal, ModalHeader, ModalBody, ModalFooter, Button } from "reactstrap";
+import PropTypes from "prop-types";
+import { updateEvent } from "../../../screens/Events/eventHelpers";
+
+const EventUnRegisterModal = ({ open, toggle, eventData, userId }) => {
+  const [isDeleting, setDeleting] = useState(false);
+
+  const handleSubmit = () => {
+    setDeleting(true);
+    onUnregisterClicked(eventData)
+      .then(() => {
+        toggle();
+        setDeleting(false);
+      })
+      .catch(console.log);
+  };
+
+  const onUnregisterClicked = async (event) => {
+    const changedEvent = {
+      // remove current user id from event volunteers
+      ...event,
+      volunteers: event.volunteers.filter((volunteer) => volunteer !== userId),
+    };
+    await updateEvent(changedEvent);
+  };
+
+  return (
+    <Modal isOpen={open} toggle={toggle} backdrop="static">
+      <ModalHeader toggle={toggle}>Delete Entry</ModalHeader>
+      <ModalBody>
+        Are you sure you want to cancel your registration for this event?
+      </ModalBody>
+      <ModalFooter>
+        <Button color="secondary" onClick={toggle}>
+          No, keep it
+        </Button>
+        <Button color="primary" onClick={handleSubmit} disabled={isDeleting}>
```

Comment:
Please update all the variables like isDeleting and text like Delete Entry so they make sense for the unregister flow